packages/expo-haptics/android/src/main/java/expo/modules/haptics/HapticsModule.kt

```diff
@@ -4,42 +4,45 @@ import android.content.Context
 import android.os.Build
 import android.os.VibrationEffect
 import android.os.Vibrator
-import expo.modules.core.ExportedModule
-import expo.modules.core.Promise
-import expo.modules.core.interfaces.ExpoMethod
-import expo.modules.haptics.arguments.HapticsInvalidArgumentException
 import expo.modules.haptics.arguments.HapticsImpactType
+import expo.modules.haptics.arguments.HapticsInvalidArgumentException
 import expo.modules.haptics.arguments.HapticsNotificationType
 import expo.modules.haptics.arguments.HapticsSelectionType
 import expo.modules.haptics.arguments.HapticsVibrationType
+import expo.modules.kotlin.Promise
+import expo.modules.kotlin.exception.Exceptions
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
 
-class HapticsModule internal constructor(context: Context) : ExportedModule(context) {
-  private val mVibrator = context.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
-  override fun getName() = "ExpoHaptics"
+class HapticsModule : Module() {
+  private val context: Context
+    get() = appContext.reactContext ?: throw Exceptions.ReactContextLost()
+  private val mVibrator get() = context.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
```

Comment:
- We should be consistent about naming. In `context` we don't use the `mCamelCase` prefix, so we can do the same here.
- IMO it looks better when the getter is in a new line
```suggestion
  private val vibrator
    get() = context.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
```